### PR TITLE
perf: change SymbolTable to stacked, removed rc

### PR DESCRIPTION
**Pull request by A4-Tacks**
Suggest using additional test cases and benchmark tests to test their correctness and performance differences first

### CODE REVIEW

**Comment by A4-Tacks** [comment]
At present, there are some bugs with it. Let's revert it for now

File: src/backend/mod.rs [hunk 1]
```diff
@@ -150,29 +158,24 @@ impl Interpreter {
...
-                    match error {
-                        Error::Break => return Ok(CrValue::Void),
-                        Error::Continue => break,
-                        _ => return Err(error),
+        self.with_block(|this| {
+            for number in (start..end).step_by(step) {
+                let number = IBig::from(number);
+                let value = Symbol::Const(variable.to_owned(), CrValue::Number(number));
+                this.symbol_tables.insert_sym(value);
+                this.symbol_tables.clear_last();
```
Comment (A4-Tacks) [inline src/backend/mod.rs:166]: Bug fix from #3

File: src/backend/scope.rs [hunk 2]
```diff
@@ -47,125 +47,147 @@ impl Symbol {
...
+        self.symbol_list_mut(id).map(|vec| vec[index] = value)
+    }
+
+    #[inline]
+    pub fn symbol_list_remove(&mut self, id: usize, index: usize) -> Result<CrValue> {
+        self.get_var_mut(id, |sym| {
+            sym.and_then(Symbol::get_value_mut)
+                .and_then(CrValue::as_list_mut)
+                .map(|list| list.remove(index))
+        })
```
Comment (A4-Tacks) [inline src/backend/scope.rs:191]: What was I thinking at the time?
It seems like we should use `symbol_list_mut` here

Comment (wenxuanjun) [reply src/backend/scope.rs:191]: Welcome to make a new pr!

**Pull request merged**